src/api/services/fetch/fetchService.ts

```diff
@@ -0,0 +1,90 @@
+import type { Config } from "../../config/type";
+import { createGitHubClient } from "../../infra/gitHub/gitHubClient";
+import type { GitHubTree } from "../../infra/gitHub/types";
+import type { Cohort, Member, Submission } from "../common/types";
+
+export function createFetchService(config: Config) {
+  const gitHubClient = createGitHubClient(config.gitHub);
+
+  return {
+    fetchMembers: async (): Promise<Member[]> => {
+      const teamNames = await gitHubClient.getTeamNames(
+        config.study.organization,
+      );
+
+      const members = await Promise.all(
+        teamNames
+          .filter((name) => name.startsWith(config.study.teamPrefix))
+          .map(async (teamName) => {
+            const members = await gitHubClient.getTeamMembers(
+              config.study.organization,
+              teamName,
+            );
+            const cohort = parseCohort(teamName, config.study.teamPrefix);
+
+            return members.map(
+              (member): Member => ({
+                id: member.login.toLocaleLowerCase(),
+                name: member.login,
+                profileUrl: member.avatar_url,
+                cohort,
+              }),
+            );
+          }),
+      );
+
+      return dropDuplicateMembers(members.flat());
+    },
+
+    fetchSubmissions: async (repoName: string): Promise<Submission[]> => {
+      const submissions = await gitHubClient.getDirectoryTree(
+        config.study.organization,
+        repoName,
+        config.study.branchName,
+      );
+
+      return submissions
+        .filter(isRelevantTree)
+        .map(parseSubmissionTree)
+        .filter((submission): submission is Submission => submission !== null);
+    },
+  };
+}
+
+const parseCohort = (teamName: string, prefix: string): Cohort => {
+  const cohort = parseInt(teamName.replace(prefix, ""), 10);
+  if (!isCohort(cohort)) {
+    throw new Error(`Invalid cohort number: ${cohort} from team ${teamName}`);
+  }
+  return cohort as Cohort;
```

Comment:
`Cohort`가 이제 단순히 `number`의 닉네임이니 type assertion은 불필요하고 코드를 읽는 분께 혼선의 여지를 남길 것 같습니다.
```suggestion
  return cohort;
```